app/views/browse/service.html

```diff
@@ -64,23 +64,29 @@
                       <dd>{{service.spec.clusterIP}}</dd>
                       <dt>Session affinity:</dt>
                       <dd>{{service.spec.sessionAffinity}}</dd>
-                      <dt ng-if="resource.status.loadBalancer.ingress.length">Ingress points</dt>
-                      <dd ng-if="resource.status.loadBalancer.ingress.length">
-                        <span ng-repeat="ingress in resource.status.loadBalancer.ingress"
+                      <dt ng-if="service.status.loadBalancer.ingress.length">Ingress Points:</dt>
+                      <dd ng-if="service.status.loadBalancer.ingress.length">
+                        <span ng-repeat="ingress in service.status.loadBalancer.ingress"
                           >{{ingress.ip}}<span ng-if="!$last">, </span></span>
                       </dd>
-                      <dt>Routes:</dt>
-                      <dd>
+                      <dt ng-if="service.spec.externalIPs.length">External IPs:</dt>
+                      <dd ng-if="service.spec.externalIPs.length">
+                        <span ng-repeat="externalIP in service.spec.externalIPs"
+                          >{{externalIP}}<span ng-if="!$last">, </span></span>
+                      </dd>
+                      <dt ng-if="!routesForService.length">Routes:</dt>
+                      <dd ng-if="!routesForService.length">
                         <span ng-if="!routesForService.length">
                           <a ng-href="project/{{project.metadata.name}}/create-route?service={{service.metadata.name}}"  ng-if="'routes' | canI : 'create'">Create route</a>
                         </span>
-                        <span ng-repeat="route in routesForService">
-                            <span ng-if="route | isWebRoute"><a ng-href="{{route | routeWebURL}}">{{route | routeLabel}}</a></span>
-                            <span ng-if="!(route | isWebRoute)">{{route | routeLabel}}</span>
-                            <span ng-show="!$last">, </span>
-                        </span>
                       </dd>
                     </dl>
+                    <div ng-if="routesForService.length" class="table-responsive small service-table">
+                      <routes-table routes="routesForService" services="services" custom-name-header="routesForService[0].kind"></routes-table>
+                    </div>
+                    <div ng-if="podsForService.length" class="table-responsive small service-table">
+                      <pods-table pods="podsForService" active-pods="podsWithEndpoints" custom-name-header="podsForService[0].kind"></pods-table>
```

Comment:
switching to LabelSelector.select is going to return a map, I would just explicitly set custom-name-header to "Pod" and don't worry about trying to use the kind here